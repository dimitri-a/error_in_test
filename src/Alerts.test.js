import React from 'react';
import { shallow } from 'enzyme';
import AlertsComp from './Alerts';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from "enzyme";

let dateNowSpy;

describe('Alerts', () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });


  it('should render pagination', () => {
    const wrapper = shallow(<AlertsComp people={peopleMock} />)
    expect(wrapper).toMatchSnapshot();
  });

});


const peopleMock = {
  "alerts": [
    {
      "submissionId": "SUB200620150004197",    
      "createdDateTime": "2018-10-24T09:25:00Z",
      "assignedDateTime": "2018-10-25T09:25:00Z",
      "dueDateTime": "2018-10-25T09:25:00Z",
      "applicants": ["000001297"],
      "id": "19d0da63-dfd0-4c00-a13a-cc822fc81297"
    },

    {
     
      "submissionId": "SUB200620150004201",
      "createdDateTime": "2018-10-24T09:25:00Z",
      "assignedDateTime": "2018-10-25T09:25:00Z",
      "dueDateTime": "2018-10-25T09:25:00Z",
      "applicants": ["000001201"],
      "id": "19d0da63-dfd0-4c00-a13a-cc822fc81201"
    },
    {
     
      "submissionId": "SUB200620150004202",
      "createdDateTime": "2018-10-24T09:25:00Z",
      "assignedDateTime": "2018-10-25T09:25:00Z",
      "dueDateTime": "2018-10-25T09:25:00Z",
      "applicants": ["000001202"],
      "id": "19d0da63-dfd0-4c00-a13a-cc822fc81202"
    },
    {
     
      "submissionId": "SUB200620150004203",
      "createdDateTime": "2018-10-24T09:25:00Z",
      "assignedDateTime": "2018-10-25T09:25:00Z",
      "dueDateTime": "2018-10-25T09:25:00Z",
      "applicants": ["000001202"],
      "id": "19d0da63-dfd0-4c00-a13a-cc822fc81203"
    },
    {
     
      "submissionId": "SUB200620150004204",
      "createdDateTime": "2018-10-24T09:25:00Z",
      "assignedDateTime": "2018-10-25T09:25:00Z",
      "dueDateTime": "2018-10-25T09:25:00Z",
      "applicants": ["000001204","000001202"],
      "id": "19d0da63-dfd0-4c00-a13a-cc822fc81204"
    },
    {
     
      "submissionId": "SUB200620150004205",
      "createdDateTime": "2018-10-24T09:25:00Z",
      "assignedDateTime": "2018-10-25T09:25:00Z",
      "dueDateTime": "2018-10-25T09:25:00Z",
      "applicants": ["000001297"],
      "id": "19d0da63-dfd0-4c00-a13a-cc822fc81205"
    }
  ]
  ,


  applicants:
    [
      {
        firstName: 'Ron',
        lastName: 'Sharma',
        idScheme: 'CustomerInternalId',
        isPrimaryApplicant: true,
        applicantSerialNumber: 1,
        id: '000001297',
      },
      {
        firstName: 'JAMESON',
        lastName: 'Last',
        idScheme: 'CustomerInternalId',
        isPrimaryApplicant: false,
        applicantSerialNumber: 2,
        id: '000001204',
      },
      {
        firstName: 'Jack',
        lastName: 'Pagoda',
        idScheme: 'CustomerInternalId',
        isPrimaryApplicant: false,
        applicantSerialNumber: 3,
        id: '000001202',
      },
      {
        firstName: 'Margot',
        lastName: 'Pagoda',
        idScheme: 'CustomerInternalId',
        isPrimaryApplicant: false,
        applicantSerialNumber: 3,
        id: '000001201',
      },
    ]
}
