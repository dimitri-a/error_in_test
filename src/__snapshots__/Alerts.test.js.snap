// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts should render pagination 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alerts
    people={
      Object {
        "alerts": Array [
          Object {
            "applicants": Array [
              "000001297",
            ],
            "assignedDateTime": "2018-10-25T09:25:00Z",
            "createdDateTime": "2018-10-24T09:25:00Z",
            "dueDateTime": "2018-10-25T09:25:00Z",
            "id": "19d0da63-dfd0-4c00-a13a-cc822fc81297",
            "submissionId": "SUB200620150004197",
          },
          Object {
            "applicants": Array [
              "000001201",
            ],
            "assignedDateTime": "2018-10-25T09:25:00Z",
            "createdDateTime": "2018-10-24T09:25:00Z",
            "dueDateTime": "2018-10-25T09:25:00Z",
            "id": "19d0da63-dfd0-4c00-a13a-cc822fc81201",
            "submissionId": "SUB200620150004201",
          },
          Object {
            "applicants": Array [
              "000001202",
            ],
            "assignedDateTime": "2018-10-25T09:25:00Z",
            "createdDateTime": "2018-10-24T09:25:00Z",
            "dueDateTime": "2018-10-25T09:25:00Z",
            "id": "19d0da63-dfd0-4c00-a13a-cc822fc81202",
            "submissionId": "SUB200620150004202",
          },
          Object {
            "applicants": Array [
              "000001202",
            ],
            "assignedDateTime": "2018-10-25T09:25:00Z",
            "createdDateTime": "2018-10-24T09:25:00Z",
            "dueDateTime": "2018-10-25T09:25:00Z",
            "id": "19d0da63-dfd0-4c00-a13a-cc822fc81203",
            "submissionId": "SUB200620150004203",
          },
          Object {
            "applicants": Array [
              "000001204",
              "000001202",
            ],
            "assignedDateTime": "2018-10-25T09:25:00Z",
            "createdDateTime": "2018-10-24T09:25:00Z",
            "dueDateTime": "2018-10-25T09:25:00Z",
            "id": "19d0da63-dfd0-4c00-a13a-cc822fc81204",
            "submissionId": "SUB200620150004204",
          },
          Object {
            "applicants": Array [
              "000001297",
            ],
            "assignedDateTime": "2018-10-25T09:25:00Z",
            "createdDateTime": "2018-10-24T09:25:00Z",
            "dueDateTime": "2018-10-25T09:25:00Z",
            "id": "19d0da63-dfd0-4c00-a13a-cc822fc81205",
            "submissionId": "SUB200620150004205",
          },
        ],
        "applicants": Array [
          Object {
            "applicantSerialNumber": 1,
            "firstName": "Ron",
            "id": "000001297",
            "idScheme": "CustomerInternalId",
            "isPrimaryApplicant": true,
            "lastName": "Sharma",
          },
          Object {
            "applicantSerialNumber": 2,
            "firstName": "JAMESON",
            "id": "000001204",
            "idScheme": "CustomerInternalId",
            "isPrimaryApplicant": false,
            "lastName": "Last",
          },
          Object {
            "applicantSerialNumber": 3,
            "firstName": "Jack",
            "id": "000001202",
            "idScheme": "CustomerInternalId",
            "isPrimaryApplicant": false,
            "lastName": "Pagoda",
          },
          Object {
            "applicantSerialNumber": 3,
            "firstName": "Margot",
            "id": "000001201",
            "idScheme": "CustomerInternalId",
            "isPrimaryApplicant": false,
            "lastName": "Pagoda",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
